tests/sympc/protocol/aby3/aby3_test.py

```diff
@@ -83,3 +85,105 @@ def test_invalid_mpc_pointer(get_clients) -> None:
     # passing sharetensor pointer
     with pytest.raises(ValueError):
         ABY3.truncate(x, session, 2 ** 32, None)
+
+
+@pytest.mark.parametrize("security_type", ["semi-honest", "malicious"])
+def test_bit_injection_prime(get_clients, security_type) -> None:
+    parties = get_clients(3)
+    falcon = Falcon(security_type=security_type)
+    session = Session(parties=parties, protocol=falcon)
+    SessionManager.setup_mpc(session)
+    ring_size = PRIME_NUMBER
+
+    bin_sh = torch.tensor([[1, 1], [0, 0]], dtype=torch.bool)
+
+    shares = [bin_sh, bin_sh, bin_sh]
+    ptr_lst = ReplicatedSharedTensor.distribute_shares(shares, session, ring_size=2)
+    x = MPCTensor(shares=ptr_lst, session=session, shape=bin_sh.shape)
+
+    xbit = ABY3.bit_injection(x, session, ring_size)
+
+    ring0 = int(xbit.share_ptrs[0].get_ring_size().get_copy())
+    result = xbit.reconstruct(decode=False)
+    exp_res = x.reconstruct(decode=False).type(torch.uint8)
```

Comment:
Q: Could you hard code the expected result? (using `x.reconstruct` causes us to rely on `reconstruct` to be correctly implemented)

Reply:
Sure George :+1:

Reply:
Also for the other tests